### ISSUE TITLE
TLS pins are not working

### ISSUE THREAD

**Issue by fmarier** [brave/brave-browser#15667]
Preloaded TLS pins are not working in Brave. For example https://pinning-test.badssl.com/ loads just fine instead of being blocked with a TLS error.

That's due to the fact that non-Chrome builds have pinning disabled: https://source.chromium.org/chromium/chromium/src/+/master:net/http/transport_security_state.cc;l=760-763;drc=735b94b59fae0115fc1b9fa736fea465e18f6bfb

While we don't want to enforce pin the pins that ship with Chromium, we do want to enforce pins on the Brave domains.

**Comment by stephendonner** [comment]
Verified `PASSED` using the testplan in https://github.com/brave/brave-core/pull/8750 with build 

Brave | 1.27.33 Chromium: 91.0.4472.77&nbsp;(Official Build)&nbsp;nightly&nbsp;(x86_64)
-- | --
Revision | 1cecd5c8a856bc2a5adda436e7b84d8d21b339b6-refs/branch-heads/4472@{#1246}
OS | macOS Version 11.4 (Build 20F71)

Steps:

1. Opened `https://pinning-test.badssl.com/` and confirmed that the page was blocked with a non-bypassable TLS error.
2. Opened a new-tab page and scroll down into the Brave Today section.
3. Confirmed that Brave Today images are loading.
4. Opened `https://creators.brave.com/` and confirmed that the page loaded.
5. Opened `https://ads-serve.brave.com/` and confirmed that the page loaded without a TLS error.

https://pinning-test.badssl.com | Brave Today | https://creators.brave.com | https://ads-serve.brave.com
--------------------------------|--------------|---------------------------|------------------------------
<img width="906" alt="Screen Shot 2021-06-01 at 1 52 26 PM" src="https://user-images.githubusercontent.com/387249/120389829-cc8aa300-c2e1-11eb-8cc2-e1ebcca9c417.png"> | <img width="906" alt="Screen Shot 2021-06-01 at 1 52 50 PM" src="https://user-images.githubusercontent.com/387249/120389853-d3b1b100-c2e1-11eb-889d-3d7a46e06143.png"> | <img width="906" alt="Screen Shot 2021-06-01 at 1 53 06 PM" src="https://user-images.githubusercontent.com/387249/120389891-e1673680-c2e1-11eb-87e4-7ac1ef7df705.png"> | <img width="906" alt="Screen Shot 2021-06-01 at 1 53 16 PM" src="https://user-images.githubusercontent.com/387249/120389895-e1ffcd00-c2e1-11eb-9cb5-43098a7acafb.png">

**Comment by GeetaSarvadnya** [comment]
@fmarier https://pinning-test.badssl.com/ fails in Brave beta (1.27.x). It works fine in Chrome. I have reset the IP (using `netsh int ip reset` and `netsh winsock reset`) and tried again but no luck.
![image](https://user-images.githubusercontent.com/38657976/124170348-ee379f80-dac4-11eb-9337-f2b4c0ceb5bf.png)
![image](https://user-images.githubusercontent.com/38657976/124170562-36ef5880-dac5-11eb-96e1-d80fe0793c4c.png)

**Comment by fmarier** [comment]
I've tested Beta 1.27.87 on Linux and Windows and I get the same error page as on Chrome in both cases. Do you see the same certificate as me on your machine?
![Screenshot from 2021-07-02 12-57-41](https://user-images.githubusercontent.com/167821/124322880-21b21280-db35-11eb-8bd8-8ba211a75212.png)

**Comment by GeetaSarvadnya** [comment]
@fmarier Thanks for checking in your machine. I have verified the certificate and found that `issued by` was showing as AVG, AVG has got installed unknowingly in my machine, which was causing this problem. I just uninstalled and tried again it worked fine. Thanks!

![image](https://user-images.githubusercontent.com/38657976/124438753-393c0600-dd96-11eb-8c43-0b63c288e842.png)

After uninstalling AVG:
![image](https://user-images.githubusercontent.com/38657976/124438895-612b6980-dd96-11eb-902f-a136d29fd945.png)
